File: problems/2639_find_the_width_of_columns_of_a_grid/Solution.java
class Solution {
    public int[] findColumnWidth(int[][] grid) {
        int m = grid.length, n = grid[0].length;
        int[] ans = new int[n];
        for (int c = 0; c < n; c++)
            for (int r = 0; r < m; r++) ans[c] = Math.max(ans[c], String.valueOf(grid[r][c]).length());
        return ans;
    }
}

File: problems/2639_find_the_width_of_columns_of_a_grid/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    vector<int> findColumnWidth(vector<vector<int>>& grid) {
        int m = grid.size(), n = grid[0].size();
        vector<int> ans(n, 0);
        for (int c = 0; c < n; c++)
            for (int r = 0; r < m; r++) ans[c] = max(ans[c], (int)to_string(grid[r][c]).size());
        return ans;
    }
};
